cloud/org.go

```diff
@@ -207,13 +209,21 @@ func (c *Client) GuessOrgMembership(ctx context.Context) (orgName, orgID string,
 	if err != nil {
 		return "", "", err
 	}
-	if len(orgs) == 0 {
+	switch len(orgs) {
+	case 0:
 		return "", "", errors.New("not a member of any organizations - cloud features require you are a member of an organization")
+	case 1:
+		return orgs[0].Name, orgs[0].ID, nil
+	case 2:
+		for _, o := range orgs {
+			if !o.Personal {
+				return o.Name, o.ID, nil
+			}
```

Comment:
I know it would be backwards incompatible, but I would have thought that it defaults to personal generally (if we ignore the history about how we got here).

I can't think of a better solution, though, the alternative is pretty bad too, because builds just start failing all of a sudden due to the missing `--org` param.

So I guess what you have here makes sense. Can we perhaps display a warning when this happens? And say that we are deprecating this behavior in the future?

Reply:
Doesn't that mean we're deprecating it now? Deprecated behaviors are behaviors that still work but are planned to be removed in the future.

Reply:
Yes, correct :+1:

Reply:
Thinking a bit more about this - should we also prevent a personal org from being auto-chosen if n = 1? I think this makes things more consistent when going to n = 2.

Reply:
I thought the personal org at n=1 is convenient at least, but agree it's a bit of a trivial choice on what makes sense when n>=2.

Are you suggesting then that in all cases, the `--org` flag or `EARTHLY_ORG` env var should be set? With the exception of keeping it compatible temporarily when n=2?

Reply:
I think so. Open to other ideas.

It's definitely annoying to have to be so verbose all the time, but it's the most clear behavior.

Perhaps the way to counteract this verbosity is to have a way to "select" an org too, similarly to how you can select satellites.

Reply:
I'm on board with the org select idea. We also don't auto-select a satellite when you have only one satellite.

That being said, I don't have time to work on org select now before free tier launches, so perhaps we just stick with the warning for compatibility and add org select later.

Reply:
SGTM!

Reply:
![Uploading Screen Shot 2023-06-30 at 3.18.48 PM.png…]()

Reply:
<img width="648" alt="Screen Shot 2023-06-30 at 3 18 48 PM" src="https://github.com/earthly/earthly/assets/3247216/ad49bfbe-f316-49c0-86b5-b03463eef695">